actions.json

```diff
@@ -0,0 +1,20 @@
+[{
+        "id": "move_all_libraries",
+        "name": "Move user's libraries",
+        "command": "/opt/yunohost/seafile/seafile_api_loader.sh /opt/yunohost/seafile/move_library.py $YNH_ACTION_USER_TO_MIGRATE",
+        "user": "root",
+        "accepted_return_codes": [0],
+        "description": {
+            "en": "WARNING: Only use this feature if you know what you do.\n This action is only needed if you have installed Seafile before the version 7 AND your Seafile instance was installed under other domain than the user accounts. Move all user's libraries to the correct account, this action exists only for handling old created account with the old SSO behavior.\n See https://github.com/YunoHost-Apps/seafile_ynh/issues/44 for more information."
+        },
+        "arguments": [
+            {
+                "name": "user_to_migrate",
+                "type": "user",
+                "ask": {
+                    "en": "User account to be migrate from the old to the new behavior"
+                },
+                "example": "bob@ynh.local"
+            }
```

Comment:
My idea was to a one argument for the old email and one other for the new email to have more flexibility.

Reply:
We can but does it make sense because the admin user need to connect to the Seafile Admin interface and then check manually the list in Local Users, then copy paste the from and the target email. As discuss with @Psycojoker `actions.json` allow defining values for select only as literal in the `actions.json` itself.

1. My script check if the user selected have a local account in Seafile (wrongly created with SSO)
1. Then it retrieve all the possible target user with username@seafile.domain in the Yunohost LDAP through the LDAP api in Seafile
1. If there is more than 1 raise an error
1. If there there is less than 1 raise an error
1. If one, import the user from LDAP to LDAPImport in Seafile
1. Change the owner of all the user's library to the new one (This operation doesn't change library id)

Reply:
Ok.

One question, does your script migrate the settings of the old user (by example language, wiki enabled, etc).

Reply:
Nop, I will check this, and we might want to delete the local user in Seafile after the migration, to ensure the migrated users do not try to connect to the old account.

Reply:
It's why I think that migrating the account instead of migrating the library from an account to an other is better.

Reply:
I check [Wiki](https://help.seafile.com/en/wiki/manage_library_as_wiki.html) are libraries ? or I missing something ?

Reply:
> I check [Wiki](https://help.seafile.com/en/wiki/manage_library_as_wiki.html) are libraries ? or I missing something ?

Well, In seafile v6 it was wiki and was renamed to published libraries in seafile v7.

It's a feature that give the possibility to set a library as a wiki. And when it's enabled you can manage your markdown file as a wiki.

Reply:
Anyway. The problem is that you can't just migrate all library there also many settings that we also want to keep on the new account.

Reply:
Ok so we have
* Avatar
* Language selection
* Name (but this is sync by LDAP)

All the other stuff are migrated through my process.
I check the Seafile API, there is no entry point for editing Avatar and Language.